### PR TITLE
Improve testing infrastructure + various refactors

### PR DESCRIPTION
**Pull request by Jules-Bertholet**
Just a little dip into the codebase, so I can familiarize myself with things. Adds CI for tests+formatting+docs+clippy, and rewrites one test in Rust so `cargo test` can run it.

### CODE REVIEW

File: tests/diff_tests.rs
Comment (noahcape) [inline tests/diff_tests.rs:1]: Thanks for making this a Rust test.

**Review by noahcape — APPROVED** [review]
This all looks great. I just pushed some changes to test so just make sure they line up. I made a small change for `pad` and `trunc` related functions and did not fix it in tests.

**Review by noahcape — APPROVED** [review]
LGTM - Thanks!

**Comment by Jules-Bertholet** [comment]
(I don't have write access, you will need to perform the merge…)

**Comment by rob-p** [comment]
Whoops, sorry about that @Jules-Bertholet. I’ve added you with write permissions.

**Pull request merged**